crates/biome_parser/src/lexer.rs

```diff
@@ -20,7 +79,7 @@ use std::iter::FusedIterator;
 ///   that any following token may turn out to be different as well, thus, it's necessary to clear the
 ///   lookahead cache.
 #[derive(Debug)]
-pub(crate) struct BufferedLexer<'l> {
+pub struct BufferedLexer<'l, Lex: Lexer<'l>> {
```

Comment:
Is it possible to keep `Lex` unbound (without the `Lexer`) and specify the bind in the `impl`?